### PR TITLE
Feat/deliverable1

### PR DESCRIPTION
**Pull request by efranco89**
## ✨ Overview 

**What?**
This is The PR for the first deliverable of the bench react bootcamp

**Why?**

**How?**
I created the following components 

Header
MainContent
Footer

MainContent has inside the Banners, Categories and FeaturedProducts components

I used font-awesome for the icons in across the app, the app was designed from mobile to desktop

**Testing?**

N/A for this first deliverable

**Screenshots**

image

image

### CODE REVIEW

File: src/Components/Banner/css/tablet.css
Comment (fervillas96) [inline src/Components/Banner/css/tablet.css:1]: Is this file needed?

File: src/Components/Banner/index.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+          />
+        )}
+      > 
+
+      </BannerCard>
+    </section>
+  )
+}
+
+export {Banner}
```
Comment (fervillas96) [inline src/Components/Banner/index.js]: This is a super NIT but, normally we use spaces between de `{}` and the words for styling purposes.
Something like `{ Banner }`.

You can download an es-lint extension for React on your visual studio environment for help whit that! (:

File: src/Components/Banner/index.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+    loading,
+    error, 
+    banners
+  } = React.useContext(BannerContext)
+  return(
+    <section className="banners">
+      <BannerCard
+        error={error}
+        loading={loading}
+        banners={banners}
```
Comment (fervillas96) [inline src/Components/Banner/index.js:17]: There is no need for this space between props inside the component definition (:

File: src/Components/BannerCard/index.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import React from "react";
+
+const BannerCard = (props) => {
```
Comment (fervillas96) [inline src/Components/BannerCard/index.js]: NIT: Props destructuring 🚀

File: src/Components/Header/index.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+import './css/desktop.css'
+
+import logo from './img/logo.png'
+
+const Header = () => {
+  return(
+    <header className="header">
+
+      <div className="header__logo">
+        {/* <img src={logo}></img> */}
```
Comment (fervillas96) [inline src/Components/Header/index.js]: NIT: Remove this comment and the `logo.png` import or use it (:

File: src/Components/Header/index.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        <ul className="header__nav-list">
+          <li className="header__nav-item">
+            <ShoppingCartIcon className="nav-cart-icon" />
+          </li>
+          <li className="header__nav-item">
+              <img
+                className="nav-account-img" 
+                src={"https://lh3.googleusercontent.com/ogw/AGvuzYbYN3iKpi6pZVDNSBfh3ysqX6LNAIuovaKmAupW6Q=s32-c-mo"}
+              ></img>
+            {/* <div className="nav-account-container">
```
Comment (fervillas96) [inline src/Components/Header/index.js]: NIT: Remove this comment

File: src/Components/Loading/index.js [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import './css/style.css'
```
Comment (fervillas96) [inline src/Components/Loading/index.js]: 💯 NICE! You create a 'skeleton' like a component for loading states!

File: src/Components/MainContent/css/style.css [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+  overflow-y: scroll;
+  overflow-x: scroll;
+  width: 100%;
+}
+
+.main__container {
+  width: 100%;
+}
+
+/* banners */
```
Comment (fervillas96) [inline src/Components/MainContent/css/style.css]: NIT: Remove unnecessary comments

File: src/Components/MainContent/index.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+const MainContent = () => {
+  
+  return(
+    <main className="main">    
+      <section className="main__container">
+        <BannerProvider>
+          <Banner />
+        </BannerProvider>
+        {/* <section className="banners">
```
Comment (fervillas96) [inline src/Components/MainContent/index.js]: NIT: Remove comments

File: src/Components/MainContent/index.js [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+import './css/style.css'
+import './css/tablet.css'
+import './css/desktop.css'
+
+const MainContent = () => {
+  
+  return(
+    <main className="main">    
+      <section className="main__container">
+        <BannerProvider>
```
Comment (fervillas96) [inline src/Components/MainContent/index.js]: Normally the Providers are imported at the `App.js` level and wrap the whole application! 
Maybe you can move the Providers to the App.js To have all of them like this:

```
<BannerProvider>
  <CategoryProvider>
    <ProductProvider>
      <Header />
      <Footer />
    </ProductProvider>
  </CategoryProvider>
</BannerProvider>
```

**Comment by fervillas96** [comment]
Good job in general!
Creating all of this in one week is amazing!

I left some minor NITs comments but nothing major! 

AMAZING WORK! 🚀

File: src/Components/Banner/css/tablet.css
Comment (efranco89) [reply src/Components/Banner/css/tablet.css:1]: Deleted

File: src/Components/Banner/index.js (hunk 1 above)
Comment (efranco89) [reply src/Components/Banner/index.js]: eslint and prettier added and run

File: src/Components/Banner/index.js (hunk 2 above)
Comment (efranco89) [reply src/Components/Banner/index.js:17]: Fixed with the lintern
